src/main/java/com/snowflake/kafka/connector/internal/streaming/schemaevolution/iceberg/IcebergFieldNode.java

```diff
@@ -61,26 +68,145 @@ StringBuilder buildQuery(StringBuilder sb, String parentType) {
     return sb;
   }
 
-  private StringBuilder appendNameAndType(StringBuilder sb) {
+  /**
+   * Method does not modify, delete any existing nodes and its types, names. It is meant only to add
+   * new children.
+   */
+  void merge(IcebergFieldNode modifiedNode) {
+    modifiedNode.children.forEach(
+        (key, node) -> {
+          IcebergFieldNode thisChild = this.children.get(key);
+          if (thisChild == null) {
+            this.children.put(key, node);
+          } else {
+            thisChild.merge(node);
+          }
+        });
```

Comment:
what if we have nodes on different lvl with same name, how we know where we should add it?

Reply:
and as this is called recursive, we might end up with adding children to many nodes on different levels